ddcommon-ffi/src/vec.rs

```diff
@@ -74,9 +73,7 @@ impl<T> From<alloc::vec::Vec<T>> for Vec<T> {
 
 impl From<anyhow::Error> for Vec<u8> {
     fn from(err: anyhow::Error) -> Self {
-        let mut vec = vec![];
-        write!(vec, "{err}").expect("write to vec to always succeed");
-        Self::from(vec)
+        Self::from(err.to_string().into_bytes())
```

Comment:
For whatever reason, clippy did not complain about this `.expect`. Rather than allow it, I figured I'd just rewrite it to a version that does the same thing. Sure, there's still a panic, but if we're ever going to audit this stuff, we'd have to audit `to_string()` usage as well.